app/src/main/java/com/beemdevelopment/aegis/ui/views/ImportEntriesAdapter.java

```diff
@@ -0,0 +1,88 @@
+package com.beemdevelopment.aegis.ui.views;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.beemdevelopment.aegis.R;
+import com.beemdevelopment.aegis.db.DatabaseEntry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+public class ImportEntriesAdapter extends RecyclerView.Adapter<ImportEntryHolder> {
+    private ArrayList<DatabaseEntry> _entries;
+    private List<ImportEntryHolder> _holders;
+
+
+    public ImportEntriesAdapter() {
+        _entries = new ArrayList<>();
+        _holders = new ArrayList<>();
+    }
+
+    public void addEntry(DatabaseEntry entry) {
+        _entries.add(entry);
+
+        int position = getItemCount() - 1;
+        if (position == 0) {
+            notifyDataSetChanged();
+        } else {
+            notifyItemInserted(position);
+        }
+    }
+
+    @Override
+    public ImportEntryHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_import_entry, parent, false);
+        return new ImportEntryHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(ImportEntryHolder holder, int position) {
+        holder.setData(_entries.get(position));
+
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                holder.toggleCheckbox();
+            }
+        });
+
+        _holders.add(holder);
```

Comment:
I don't think it's a good idea to use a ViewHolder list to keep track of checked items. They're never removed from the list, so if there's a long list of entries and the user scrolls around a bit, they'll end up with a bunch of duplicates. The list will also be incomplete if the user hasn't scrolled down all the way yet.

A simple solution could be to create a new class with a field for the DatabaseEntry and a 'checked' field. Pass that to the ViewHolders instead.